checks/binary_artifact.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2021 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checks
+
+import (
+	"fmt"
+
+	"github.com/h2non/filetype"
+	"github.com/h2non/filetype/types"
+
+	"github.com/ossf/scorecard/checker"
+)
+
+//nolint
+func init() {
+	registerCheck(CheckBinaryArtifacts, BinaryArtifacts)
+}
+
+const CheckBinaryArtifacts string = "Binary-Artifacts"
+
+// BinaryArtifacts  will check the repository if it contains binary artifacts.
+func BinaryArtifacts(c *checker.CheckRequest) checker.CheckResult {
+	return CheckFilesContent(CheckBinaryArtifacts, "*", false, c, checkBinaryFileContent)
+}
+
+func checkBinaryFileContent(path string, content []byte,
+	logf func(s string, f ...interface{})) (bool, error) {
+	binaryFileTypes := map[string]bool{
+		"crx": true, "deb": true, "dex": true, "dey": true,
+		"elf": true, "bin": true, "o": true, "so": true, "iso": true, "class": true, "jar": true,
+		"bundle": true, "dylib": true, "lib": true, "msi": true, "acm": true, "ax": true,
+		"cpl": true, "dll": true, "drv": true, "efi": true, "exe": true, "mui": true, "ocx": true,
+		"scr": true, "sys": true, "tsp": true, "pyc": true, "pyo": true, "par": true,
+		"rpm": true, "swf": true, "torrent": true, "cab": true,
+	}
+	var t types.Type
+	var err error
+	if t, err = filetype.Get(content); err != nil {
+		return false, fmt.Errorf("failed in getting the content type %w", err)
```

Comment:
Should we fallback to actual file extension here instead of failing. Reason is some content matchers are not in filetype e.g. iso, etc, so we should still try the direct extension.

Also, should leave a TODO to add file content matchers for those types that are not in filetype, e.g. .so, .iso.

Reply:
I agree with the suggestion of falling back to file-based extension checks. I will add that.

I am not in favor of adding TODO's because most of the time people don't get to it. I am going to open an issue for it to be solved.

Reply:
as more people contribute to it, todos should get fixed. opening an issue is fine as well.

Reply:
I am opening issues with "help wanted" so that we can attract more new contributors.

Reply:
https://github.com/ossf/scorecard/issues/596